src/main/groovy/life/qbic/utils/DataSetParser.groovy

```diff
@@ -0,0 +1,35 @@
+package life.qbic.utils
+
+/**
+ * A parser that takes an input data structure and tries to convert it to a certain type dataset
+ * of type T.
+ *
+ * The user of this interface can expect a valid dataset of type T if no exceptions have
+ * been thrown after parsing of data structure has been called.
+ *
+ * @since 1.7.0
+ */
+interface DataSetParser<T> {
```

Comment:
```suggestion
interface DataSetParser<T> {
```
Do we leave complete freedom as to the choice of `T` or do we want to box it in by extending a data type class or equivalent?

Reply:
Hm, thinking about it i don't want to make this restriction at this development stage now.

Reply:
I would also argue that it is not necessary at the moment. :+1: